packages/shell-api/src/shell-internal-state.ts

```diff
@@ -330,6 +336,51 @@ export default class ShellInternalState {
     };
   }
 
+  get interrupted() {
+    return this.isInterrupted;
+  }
+
+  async onInterruptExecution(): Promise<boolean> {
+    this.isInterrupted = true;
+    this.resumeMongosAfterInterrupt = await Promise.all(this.mongos.map(async m => {
+      try {
+        return {
+          mongo: m,
+          resume: await m._suspend()
+        };
+      } catch (e) {
+        return {
+          mongo: m,
+          resume: null
+        };
+      }
+    }));
+    return !this.resumeMongosAfterInterrupt.find(r => r.resume === null);
+  }
+
+  async onResumeExecution(): Promise<boolean> {
+    const promises = this.resumeMongosAfterInterrupt?.map(async r => {
+      if (!this.mongos.find(m => m === r.mongo)) {
+        // we do not resume mongo instances that we don't track anymore
+        return true;
+      }
+      if (r.resume === null) {
+        return false;
+      }
+      try {
+        await r.resume();
+        return true;
+      } catch (e) {
+        return false;
+      }
+    }) ?? [];
+    this.resumeMongosAfterInterrupt = undefined;
+
+    const result = await Promise.all(promises);
+    this.isInterrupted = true;
```

Comment:
Is this supposed to be `true` here? If yes, I think I need to read the code a bit more :)

Reply:
Ah no of course not! Thanks...